mteb/models/model_implementations/slm_models.py

```diff
@@ -0,0 +1,484 @@
+"""
+SauerkrautLM Visual Document Retrieval Models - MTEB Integration
+
+This module provides MTEB wrappers for SauerkrautLM ColPali-style models:
+- SLM-ColQwen3 (Qwen3-VL backbone)
+- SLM-ColLFM2 (LFM2 backbone)
+- SLM-ColMinistral3 (Ministral3 backbone)
+
+Based on:
+- MTEB ColPali implementation: mteb/models/model_implementations/colpali_models.py
+"""
+
+from __future__ import annotations
+
+import logging
+from functools import partial
+from typing import Any
+
+import torch
+from PIL import Image
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+
+from mteb._requires_package import (
+    requires_image_dependencies,
+    requires_package,
+)
+from mteb.abstasks.task_metadata import TaskMetadata
+from mteb.models.abs_encoder import AbsEncoder
+from mteb.models.model_meta import ModelMeta, ScoringFunction
+from mteb.types import Array, BatchedInput, PromptType
+
+logger = logging.getLogger(__name__)
+
+
+# =============================================================================
+# Supported Languages
+# =============================================================================
+
+SUPPORTED_LANGUAGES = [
+    "eng-Latn",  # English
+    "deu-Latn",  # German
+    "fra-Latn",  # French
+    "spa-Latn",  # Spanish
+    "ita-Latn",  # Italian
+    "por-Latn",  # Portuguese
+]
+
+
+# =============================================================================
+# Base Wrapper Class
+# =============================================================================
+
+class SLMBaseWrapper(AbsEncoder):
+    """
+    Base wrapper for SauerkrautLM multi-vector embedding models.
+    
+    All our models use late interaction (MaxSim) for retrieval scoring.
+    """
+    
+    model_class = None
+    processor_class = None
+    model_name_prefix = "SLM"
+
+    def __init__(
+        self,
+        model_name: str,
+        revision: str | None = None,
+        device: str | None = None,
+        use_flash_attn: bool = True,
+        **kwargs,
+    ):
+        requires_image_dependencies()
+        requires_package(
+            self, "sauerkrautlm_colpali", model_name, "pip install sauerkrautlm-colpali"
+        )
+        
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+        self._load_model_and_processor(model_name, revision, use_flash_attn, **kwargs)
```

Comment:
Can you change to something like? This would simplify code a bit
```suggestion
        self._load_model_and_processor(model_name, revision, use_flash_attn, **kwargs)
        self.mdl = self.mdl.to(self.device)
        self.mdl.eval()
```